app/hooks/useAuth.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react'
+import Cookies from 'js-cookie'
+import type { VerifyOptions } from 'jsonwebtoken'
+import { decode, verify } from 'jsonwebtoken'
+import { z } from 'zod'
+import { loginWithRefreshToken } from '../utils/login'
+import config from '../utils/config'
+
+// this type will be extended in the future
+const jwtUserPayload = z.object({
+  username: z.string()
+})
+
+type User = z.output<typeof jwtUserPayload>
+
+// TODO: decide on some sensible options
+const jwtDefaultOptions: VerifyOptions = {
+
+}
+
+// TODO: this will have to be set somehow; maybe fetch this at startup or bake it in at build time
+// TODO: could also use something like .well-known / jwks
+const cert = ''
+
+const jwt = {
+  parse: (jwt: string): User => {
+    const decoded = decode(jwt, jwtDefaultOptions)
+    const parsed = z.object({ user: jwtUserPayload }).safeParse(decoded)
+    if (decoded === null || !parsed.success) {
+      throw new Error('Failed to decode token')
+    }
+    return parsed.data.user
+  },
+  verify: (jwt: string): boolean => {
+    if (config.mock) {
+      return z.object({ user: jwtUserPayload }).safeParse(decode(jwt)).success
+    }
+    try {
+      const payload = verify(jwt, cert, { ...jwtDefaultOptions, complete: false })
+      return z.object({ user: jwtUserPayload }).safeParse(payload).success
+    } catch (error) {
+      return false
+    }
+  }
+}
+
+type UseAuthOptions = {
```

Comment:
Since we need the names of the cookies more often, I have centralized them here.